### ISSUE TITLE
Docker pull no longer takes a -t parameter

### ISSUE THREAD

**Issue by naftulikay** [garethr/garethr-docker#89]
I'm running Ubuntu 14.04 with latest `lxc-docker`:

```
Client version: 1.2.0
Client API version: 1.14
Go version (client): go1.3.1
Git commit (client): fa7b24f
OS/Arch (client): linux/amd64
Server version: 1.2.0
Server API version: 1.14
Go version (server): go1.3.1
Git commit (server): fa7b24f
```

When I try running the following:

```
docker::image{ 'rfkrocktk/puppetmaster':
     image_tag => '1.0.5',
}
```

The actual command which is run is `sudo docker pull -t="1.0.5" rfkrocktk/puppetmaster`. This causes Docker to download _every_ version of my Docker image (5 of them at 600MB each).

When I edit it to read the following:

```
docker::image { 'rfkrocktk/puppetmaster:1.0.5': }
```

It works properly, however it runs the Docker `pull` command each time because the `$image_find` command doesn't work properly with the given name above.

**TL;DR**: Docker got rid of the `-t` parameter to `docker pull` and now just expects the following format:

```
vagrant@vagrant-trusty64:~$ sudo docker help pull

Usage: docker pull NAME[:TAG]

Pull an image or a repository from the registry
```

I'll see if I can contribute a fix, but this is broken.
